langlib/langlib-test/src/test/resources/test-src/tablelib_test.bal

```diff
@@ -282,8 +282,40 @@ function removeIfHasKey() returns boolean {
     return removedPerson1 == () && removedPerson2?.name == "Chiran";
 }
 
-function testHasKey() returns boolean {
-    return tab.hasKey("Mohan");
+type age record {
+    int|string? age;
+};
+
+type NewPerson record {
+  string name;
+  readonly age age;
+};
+
+type NewPersonalTable table<NewPerson> key(age);
+
+function testHasKey() {
+    table<record { readonly int|string|float? k; }> key(k) tbl1 = table[];
+    tbl1.add({k: 0});
```

Comment:
Shall we also add a test case for key as a ()?

Reply:
👍🏽  Below code is crashing at the moment
```Ballerina
type J record {
    readonly string|int? k;
};

public function main() {
   table<J> key(k) t = table[];
   t.add({k: ()});
}
```

```LogTalk
[2021-09-23 14:06:39,562] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
        at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
        at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
        at io.ballerina.runtime.internal.values.TableValueImpl$KeyHashValueHolder.putData(TableValueImpl.java:615)
        at io.ballerina.runtime.internal.values.TableValueImpl$KeyHashValueHolder.addData(TableValueImpl.java:572)
        at io.ballerina.runtime.internal.values.TableValueImpl.add(TableValueImpl.java:217)
        at org.ballerinalang.langlib.table.Add.add(Add.java:38)
        at ballerina.lang$0046table.0.table.add(table.bal:104)
        at t2.main(t2.bal:8)
        at $_init.$lambda$main$(.)
        at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:612)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:325)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:293)
        at java.base/java.lang.Thread.run(Thread.java:834)
        ```

Reply:
Will look into that